File: gaia-sdk-python/graphql/response/type/Knowledge.py

from graphql.response.type.Fulfilment import Fulfilment
from graphql.response.type.Behaviour import Behaviour
from graphql.response.type.Statement import Statement
from graphql.response.type.KnowledgeEdge import KnowledgeEdge
from graphql.response.type.Intent import Intent
from graphql.response.type.Prompt import Prompt
from graphql.response.type.Code import Code


class Knowledge:
    def intents(self) -> Intent:
        return self.intents
    def prompts(self) -> Prompt:
        return self.prompts
    def fulfilments(self) -> Fulfilment:
        return self.fulfilments
    def statements(self) -> Statement:
        return self.statements
    def codes(self) -> Code:
        return self.codes
    def behaviours(self) -> Behaviour:
        return self.behaviours
    def edges(self) -> KnowledgeEdge:
        return self.edges

File: gaia-sdk-java/gaia-sdk-graphql/src/main/kotlin/gaia/sdk/response/type/Knowledge.kt
package gaia.sdk.response.type

import gaia.sdk.Uuid
import gaia.sdk.ISO8601
import gaia.sdk.Struct
import gaia.sdk.response.intf.*
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonCreator
import gaia.sdk.request.enumeration.*

data class Knowledge @JsonCreator constructor(
    @JsonProperty("users") val users:List<User>? = null,
    @JsonProperty("user") val user:User? = null,
    @JsonProperty("apiKeys") val apiKeys:List<ApiKey>?,
    @JsonProperty("apiKey") val apiKey:ApiKey?,
    @JsonProperty("roles") val roles:List<Role>? = null,
    @JsonProperty("role") val role:Role? = null,
    @JsonProperty("tenants") val tenants:List<Tenant>? = null,
    @JsonProperty("tenant") val tenant:Tenant? = null,
    @JsonProperty("identities") val identities:List<Identity>? = null,
    @JsonProperty("identity") val identity:Identity? = null,
    @JsonProperty("intents") val intents:List<Intent>? = null,
    @JsonProperty("intent") val intent:Intent? = null,
    @JsonProperty("prompts") val prompts:List<Prompt>? = null,
    @JsonProperty("prompt") val prompt:Prompt? = null,
    @JsonProperty("fulfilments") val fulfilments:List<Fulfilment>? = null,
    @JsonProperty("fulfilment") val fulfilment:Fulfilment? = null,
    @JsonProperty("statements") val statements:List<Statement>? = null,
    @JsonProperty("statement") val statement:Statement? = null,
    @JsonProperty("codes") val codes:List<Code>? = null,
    @JsonProperty("code") val code:Code? = null,
    @JsonProperty("behaviours") val behaviours:List<Behaviour>? = null,
    @JsonProperty("behaviour") val behaviour:Behaviour? = null,
    @JsonProperty("edges") val edges:List<Edge>? = null,
    @JsonProperty("edge") val edge:Edge? = null,
    @JsonProperty("skills") val skills:List<Skill>? = null,
    @JsonProperty("skill") val skill:Skill? = null,
    @JsonProperty("skillProvisions") val skillProvisions:List<SkillProvision>? = null,
    @JsonProperty("skillProvision") val skillProvision:SkillProvision? = null
)